### PR TITLE
Added edge case handling for weird IBM ldap issues

### PR DESCRIPTION
**Pull request by aidenfoxx**
Solves https://github.com/directus/directus/issues/9516

### CODE REVIEW

File: api/src/exceptions/unexpected-response.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { BaseException } from '@directus/shared/exceptions';
+
+export class UnexpectedResponseException extends BaseException {
+	constructor(message: string) {
+		super(message, 500, 'UNEXPECTED_RESPONSE');
```
Comment (rijkvanzanten) [inline api/src/exceptions/unexpected-response.ts:5]: Should we do a 503 here, as it's the external service that's unavailable?

Comment (aidenfoxx) [reply api/src/exceptions/unexpected-response.ts:5]: It's not unavailable though. The issue is that it is available when it shouldn't be. We expect a failure and never get it.

Comment (aidenfoxx) [reply api/src/exceptions/unexpected-response.ts:5]: That being said, I don't mind what we use

Comment (rijkvanzanten) [reply api/src/exceptions/unexpected-response.ts:5]: 500 is just a generic "internal" server error, but this isn't an unknown error that's happening within Directus itself, but rather an unexpected error in the related service. I agree that a 503 is technically not 100% accurate, but it feels closer to this particular situation than a 500 to me 🤔

Comment (aidenfoxx) [reply api/src/exceptions/unexpected-response.ts:5]: Sure, we can go for that then. Nobody outside of MiniDigger is ever gonna see this anyway. 😂

Comment (aidenfoxx) [inline api/src/exceptions/unexpected-response.ts:5]:
```suggestion
		super(message, 503, 'UNEXPECTED_RESPONSE');
```

Comment (MiniDigger) [reply api/src/exceptions/unexpected-response.ts]: I feel attacked! 
Sadly I don't think we are the only IBM zOS customer thats left for them to milk...

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**